e2e/cypress/integration/channel_sidebar/dm_gm_behaviour_spec.js

```diff
@@ -247,4 +247,120 @@ describe('DM category', () => {
             });
         });
     });
+
+    it('MM-T2003 Number of direct messages to show', () => {
+        const receivingUser = testUser;
+
+        // * Verify that we can see the sidebar
+        cy.get('#headerTeamName').should('be.visible');
+
+        // # Collapse the DM category (so that we can check all unread DMs quickly without the sidebar scrolling beind an issue)
+        cy.get('button.SidebarChannelGroupHeader_groupButton:contains(DIRECT MESSAGES)').should('be.visible').click();
+
+        // # Create 41 DMs (ie. one over the max displayable read limit)
+        for (let i = 0; i < 41; i++) {
```

Comment:
Is there a better way to repeat something a finite number of times with Cypress? I noticed my browser would crash here and there running these tests and I'm wondering if I'm doing something wrong...

Reply:
@nevyangelova just an fyi 
You will need something similar for [MM-T3168](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3168)

Reply:
Unfortunately, it only works for `i >= 20`, as we are filtering DMs at that point. Since that's been recalculated on the front-end, I wonder if its contributing to this issue...

Nonetheless, I've added it in for `i < 20` and I didn't run into any crashes.

Reply:
I missed to mention that during my local testing, I changed the user's preference to show all DMs and revert to default via API after all the required messages were completed.

Yeah, that's also possible. I'll keep an eye on this once merged, and see if there's anything needs to improve.

Thanks for updating, and your workaround works for me too!